### PR TITLE
Realistic biome support for BOP

### PR DESCRIPTION
**Pull request by whichonespink44**
Realistic versions of BOP biomes are here at last!

Well, nearly.

I've done everything except the surfaces, which I'm going to need a hand with.

I've done a few of the surfaces, but all I did was copy & paste from vanilla, and I could probably do that for most of the other BOP biomes, but I'd like to understand the new surfacing system so I can do it properly.

Just posting this PR so you guys can review the code and make sure I've not missed anything or done anything wrong.

### CODE REVIEW

File: src/main/java/teamrtg/rtg/modules/bop/RTGModuleBOP.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    /**
+     * Is the mod present?
+     * @return true if mod is currently installed
+     * @see ModPresenceTester
+     */
+    @Override
+    public boolean isPresent() {
+        return false;
```
Comment (whichonespink44) [inline src/main/java/teamrtg/rtg/modules/bop/RTGModuleBOP.java:30]: Do I need to add functionality to this method? Something like:

```
public boolean isPresent() {
    ModPresenceTester mpt = new ModPresenceTester("BiomesOPlenty");
    return mpt.present();
}
```

File: src/main/java/teamrtg/rtg/modules/bop/biomes/RTGBiomeBOPCherryBlossomGrove.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+        DecoFallenTree decoFallenTree1 = new DecoFallenTree();
+        decoFallenTree1.distribution.noiseDivisor = 100f;
+        decoFallenTree1.distribution.noiseFactor = 6f;
+        decoFallenTree1.distribution.noiseAddend = 0.8f;
+        decoFallenTree1.logCondition = DecoFallenTree.LogCondition.NOISE_GREATER_AND_RANDOM_CHANCE;
+        decoFallenTree1.logConditionNoise = 0f;
+        decoFallenTree1.logConditionChance = 16;
+        decoFallenTree1.maxY = 100;
+        decoFallenTree1.logBlock = BlockBOPLog.paging.getVariantState(BOPWoods.CHERRY);
```
Comment (whichonespink44) [inline src/main/java/teamrtg/rtg/modules/bop/biomes/RTGBiomeBOPCherryBlossomGrove.java:74]: There's probably a better way to access BOP's blocks than this:

```
decoFallenTree1.logBlock = BlockBOPLog.paging.getVariantState(BOPWoods.CHERRY);
decoFallenTree1.leavesBlock = BlockBOPLeaves.paging.getVariantState(BOPTrees.PINK_CHERRY);
```

In 1.7.10, we accessed the block registry and grabbed the block by its name, but I wasn't sure how to do that with BOP's wood & leaves as they seemed to be setup differently than regular blocks?

File: src/main/java/teamrtg/rtg/modules/bop/biomes/RTGBiomeBOPBayou.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        return new TerrainBase() {
+
+            @Override
+            public float generateNoise(RTGWorld rtgWorld, int x, int y, float biomeWeight, float border, float river) {
+                return terrainPlains(x, y, rtgWorld.simplex, river, 80f, 1f, 40f, 20f, 62f);
+            }
+
+            /*
+            @Override
+            public float lakePressure(OpenSimplexNoise simplex, CellNoise simplexCell, int x, int y, float border)
```
Comment (whichonespink44) [inline src/main/java/teamrtg/rtg/modules/bop/biomes/RTGBiomeBOPBayou.java:29]: I wasn't sure where to put the water feature code, so I commented it out in the terrains.

**Comment by whichonespink44** [comment]
Gonna merge this tonight with unfinished surface porting to avoid conflicts with the surface wrapping branch I'm working on.

**Comment by vidaj** [comment]
EDIT: Nevermind, it's the mod checking that's broken, not this PR.
Just a heads up, this change causes RTG to crash on startup when BOP is not present. There's no actual checking of BOP presence before using BOP classes.

Caused by: java.lang.NoClassDefFoundError: biomesoplenty/api/biome/BOPBiomes
    at teamrtg.rtg.modules.bop.biomes.RTGBiomeBOPAlps.<init>(RTGBiomeBOPAlps.java:16)
    at teamrtg.rtg.modules.bop.RealisticBOPBiomes.initBiomes(RealisticBOPBiomes.java:82)
    at teamrtg.rtg.api.module.RTGModule.initBiomes(RTGModule.java:48)
    at teamrtg.rtg.api.module.Mods.initAllBiomes(Mods.java:30)
    at teamrtg.rtg.core.RTG.postInit(RTG.java:67)

**Comment by whichonespink44** [comment]
Thanks for the heads-up @vidaj - we'll get this sorted tonight.

**Pull request merged**